MA-291 Allow API access without email activation.

nasthagiri opened the pull request:
https://openedx.atlassian.net/browse/MA-291
* This PR allows users on Mobile apps to authenticate via OAuth without having their email addresses verified.
* It also updates the enrollment API so the OAuth flow is consistent with the SessionAuth flow regarding email address verification.
* The corresponding PR on edx-oauth2-provider is https://github.com/edx/edx-oauth2-provider/pull/18 .

Please review:
@wedaly or @rlucioni 
@BenjiLee

nasthagiri commented:
test this please.

nasthagiri commented:
test this please.

BenjiLee commented on common/djangoapps/enrollment/views.py:
```diff
@@ -64,7 +63,7 @@ class EnrollmentView(APIView):
     """
 
-    authentication_classes = OAuth2Authentication, SessionAuthenticationAllowInactiveUser
+    authentication_classes = OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser
```
Just to confirm, you renamed this for clarity?

BenjiLee commented on requirements/edx/github.txt:
```diff
@@ -32,7 +32,7 @@ git+https://github.com/mitocw/django-cas.git@60a5b8e5a62e63e0d5d224a87f0b489201a
 -e git+https://github.com/edx/ease.git@97de68448e5495385ba043d3091f570a699d5b5f#egg=ease
 -e git+https://github.com/edx/i18n-tools.git@56f048af9b6868613c14aeae760548834c495011#egg=i18n-tools
--e git+https://github.com/edx/edx-oauth2-provider.git@0.4.0#egg=oauth2-provider
+-e git+https://github.com/edx/edx-oauth2-provider.git@04b4302b3f61e6e72acec41467f81ab62c59c3dd#egg=oauth2-provider
```
Note: Tests are not passing on the Oath2 repo so this may need to change.

BenjiLee commented:
LGTM

rlucioni commented:
This looks reasonable to me. There are two acceptance tests failing your build. One looks flaky, but the other is related to password reset from the login page and might be worth looking into.

jzoldak commented:
test this please

nasthagiri commented:
test this please.

jzoldak commented:
test this please

jzoldak commented:
test this please

nasthagiri commented:
@BenjiLee @rlucioni I'd like to merge this in before the release cut at 10am today.  Let me know if you have any concerns/issues with the latest changes (squashed for timing): 
1. Added test_authentication.py
2. Updated github.txt
Thanks.

rlucioni commented:
:+1:
